fix(zone.js): correct defineProperties for symbol props.

varomodt opened the pull request:
Currently, this `defineProperties` patch does not work for symbol properties which will not be enumerated by Object.keys. This CL adds a second case for symbols, with a small feature guard for IE.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

I'm unsure if this change requires a change to tests or docs. Please let me know if it does.

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?

Allows usage of Object.defineProperties with symbol properties.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

Not unless there is an application currently depending on a symbol property _not_ being applied by defineProperties.

## Other information

varomodt committed:
Fix defineProperties for symbol props.

Currently, this `defineProperties` patch does not work for symbol properties which will not be enumerated by Object.keys. This CL adds a second case for symbols, with a small feature guard for IE.

varomodt committed:
fix(zone.js): correct defineProperties for symbol props.

Currently, this `defineProperties` patch does not work for symbol
properties which will not be enumerated by Object.keys. This CL adds a
second case for symbols, with a small feature guard for IE.

varomodt committed:
Update define-property.ts

varomodt committed:
fix(zone.js): correct defineProperties for symbol props.

Currently, this `defineProperties` patch does not work for symbol
properties which will not be enumerated by Object.keys. This CL adds a
second case for symbols, with a small feature guard for IE.

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines:
  https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
  - [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] Docs have been added / updated (for bug fixes / features)

  I'm unsure if this change requires a change to tests or docs. Please
  let me know if it does.

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

<!-- Please describe the current behavior that you are modifying, or
link to a relevant issue. -->

Issue Number: N/A

Allows usage of Object.defineProperties with symbol properties.

- [ ] Yes
- [X] No

<!-- If this PR contains a breaking change, please describe the impact
and migration path for existing applications below. -->

Not unless there is an application currently depending on a symbol
property _not_ being applied by defineProperties.

varomodt committed:
fix(zone.js): correct defineProperties for symbol props.

varomodt committed:
Update define-property.ts

varomodt committed:
fix(zone.js): correct defineProperties for symbol props.

Currently, this defineProperties patch does not work for symbol
properties which will not be enumerated by Object.keys. This CL adds a
second case for symbols, with a small feature guard for IE.

JiaLiPassion committed:
fix(zone.js): update type definition of defineProperties

Update type definition to remove `any` cast

JiaLiPassion committed:
fix(zone.js): update type definition of defineProperties

Update type definition to remove `any` cast

JiaLiPassion committed:
test(zone.js): add test cases of Object.defineProperties()

Add test cases of `Object.definiProperties()` zone.js patch to test
1. defineProperties for normal string props
2. defineProperties for symbol props
3. defineProperties for normal string props + symbol props

JiaLiPassion committed:
test(zone.js): add test cases of Object.defineProperties()

Add test cases of `Object.definiProperties()` zone.js patch to test
1. defineProperties for normal string props
2. defineProperties for symbol props
3. defineProperties for normal string props + symbol props

JiaLiPassion commented:
@varomodt, thank you for the PR.
I added two commits to update the `type def` to remove the `any cast`, and also added test cases. Also this PR closes #44095

varomodt committed:
fix(zone.js): merge fix of defineProperties for symbol props.

Currently, this `defineProperties` patch does not work for symbol
properties which will not be enumerated by Object.keys. This CL adds a
second case for symbols, with a small feature guard for IE.

varomodt committed:
fix(zone.js): merge defineProperty fixes

Currently, this `defineProperties` patch does not work for symbol
properties which will not be enumerated by Object.keys. This CL adds a
second case for symbols, with a small feature guard for IE.

varomodt committed:
fix(zone.js): correct defineProperties for symbol props.

Currently, this defineProperties patch does not work for symbol
properties which will not be enumerated by Object.keys. This CL adds a
second case for symbols, with a small feature guard for IE.

JiaLiPassion reviewed (changes requested):
Hi, @varomodt 
It seems there are several commits not belong to this PR, and also several commits seems not changing files, could you check it and rebase the PR?
Also could you please update the commit message to meet the guideline here https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit, 
thank you.

JiaLiPassion commented:
@varomodt , I checked your commits, it seems there are some `aio`, `animations` commits are mixed there, do you mind I create another PR to only include the `defileProperties` fix? Thank you.
